src/instructlab/eval/mmlu.py

```diff
@@ -0,0 +1,69 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Local
+from .evaluator import Evaluator
+
+
+class MMLU_Evaluator(Evaluator):
+    """
+    Child class of an Evaluator for Massive Multitask Language Understanding (MMLU)
+
+    Attributes:
+        tasks       list of tasks for MMLU to test the model with
+        fewshots    number of examples
+        batchsize   number of GPUs
+    """
+
+    def __init__(
+        self, model_path, tasks: list[str], fewshots: int = 2, batchsize: int = 5
+    ) -> None:
+        super().__init__(model_path)
+        self.tasks = tasks
+        self.fewshots = fewshots
+        self.batchsize = batchsize
+
+    def run(self) -> dict:
+        individual_scores: dict[str, float] = {}
+        overall_score: float = 0.0
+        payload = {
+            "individual_scores": individual_scores,
+            "overall_score": overall_score,
```

Comment:
Would `return overall_score, individual_scores` be preferable?

Reply:
I think for @cdoern it would be easier to parse a dict but I'll let him make the final call there

Reply:
I think a single dict makes more sense, less return variability the better

Reply:
Why create a dict just to unpack it?  Plus you can get the type hinting with the tuple.  And wouldn't there be no "parsing" at all with a tuple?

Reply:
> less return variability the better

Is this referring to changes over time not showing up in the definition signature?  The downside though is you are moving your api into the dict keys.  The same breakages will happen on changes, but there is now a naming aspect (of the keys) that needs to be described in the api that won't be evident from the signature.

Reply:
actually, you're right I think I will just be dissecting the dict. The only thing I am wondering is if there's no constant structure, how will I know which values are which?

Reply:
> The only thing I am wondering is if there's no constant structure, how will I know which values are which?

I am not sure I am following.  The call would look something like:

`overall_score, individual_scores = eval.run(...)`

or

`overall_score = eval.run()`

if you only want the first value.  That would be consistent.  If additional return values are ever added to the end, callers would only need to be updated if they want to address the new values.

Reply:
I pushed up a possible implementation here. Let me both know your thoughts!